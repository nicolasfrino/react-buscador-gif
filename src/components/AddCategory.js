import React, {useState} from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputvalue, setinputvalue] = useState('');

    const handleInputChange = (e) => {
        setinputvalue(e.target.value);

        //console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        //console.log('handleSubmit'); 

        if (inputvalue.trim().length > 2) {
            setCategories(cats => [ inputvalue, ...cats]); //cats => ['Pokemon', ...categories]
            setinputvalue('');
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="Text"
                value={inputvalue}
                //onChange={ (e) => console.log(e)}
                onChange={ handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;