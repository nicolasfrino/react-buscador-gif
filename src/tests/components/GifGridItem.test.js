import React from 'react';
//const { shallow } = require("enzyme");
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

const title = 'un titulo';
const url = 'https://localhost/algo.jpg';
const wrapper =  shallow ( <GifGridItem title={title} url={url}  /> );
const p = wrapper.find('p');
   
    test ('debe mostrar compo OK', () => {
        expect ( wrapper ).toMatchSnapshot();

    })

    test ('debe tener parrafo con titulo', () => {
        expect ( p.text().trim() ).toBe( title );

    })

    test ('debe tener imagen con url y alt', () => {
        const img = wrapper.find('img');
        //console.log(img.props().src ); 
        //console.log(img.prop('src') ); 
        expect (img.prop('src')).toBe(url);
        expect (img.prop('alt')).toBe(title);

    })

    test ('debe tener clse', () => {
        const div = wrapper.find('div');
        console.log(div.props().className ); 
        expect (div.props().className.includes('fadeIn') ).toBe(true);

    })

})
