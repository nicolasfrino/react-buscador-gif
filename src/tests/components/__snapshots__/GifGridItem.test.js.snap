// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar compo OK 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
